foreman_stub.py

```diff
@@ -54,7 +55,10 @@ def get_hosts(hostid=None):
         cache_key = f'{cache_key}/{hostid}'
 
     try:
-        resp = PAGECACHE[cache_key][pagenum]
+        if request.path.endswith('/facts'):
```

Comment:
I'd do
```python
resp = PAGECACHE[cache_key]
if not request.path.endswith('/facts'):
    resp = resp[pagenum]
```

Reply:
Also, put the first line outside of try/except, because it cannot `KeyError` out:
```python
    ...
    resp = PAGECACHE[cache_key]
    try:
        if not request.path.endswith('/facts'):
            resp = resp[pagenum]
￼    except KeyError:
￼    ...
```